### PR TITLE
Adds windows specific handling to refcount and plugin_utils

### PR DESCRIPTION
**Pull request by venilnoronha**
## Description
This PR enables the `refcount` module to connect to the Docker host under a Windows environment, and it also enables listing of mounts.

* Moves `GetMountInfo` from `plugin_utils` to the `fs` module.
* Adds `PowerShell` based `GetMountInfo` implementation for Windows.
* Updates build script.

## Test Results
The `test-all` target passes locally and following is the tail'd output.
```
2017-07-17T23:06:29-07:00 END: Running TestSanity on  tcp://10.192.122.236:2375 (may take a while)...
--- PASS: TestSanity (6.50s)
	sanity_test.go:207: Successfully connected to tcp://10.192.122.236:2375
	sanity_test.go:207: Successfully connected to tcp://10.192.108.211:2375
	sanity_test.go:225: Creating vol=DefaultTestVol on client tcp://10.192.122.236:2375.
	sanity_test.go:111: Running cmd=&[touch /mnt/testvol/DefaultTestVol/file_to_touch] with vol=DefaultTestVol on client tcp://10.192.122.236:2375
	sanity_test.go:111: Running cmd=&[stat /mnt/testvol/DefaultTestVol/file_to_touch] with vol=DefaultTestVol on client tcp://10.192.122.236:2375
=== RUN   TestConcurrency
2017-07-17T23:06:29-07:00 Running concurrent tests on tcp://10.192.122.236:2375 and tcp://10.192.108.211:2375 (may take a while)...
2017-07-17T23:06:29-07:00 START: Running create/delete multi-host concurrent test ...
2017-07-17T23:06:39-07:00 END: Running create/delete multi-host concurrent test ...
Running same docker host concurrent create/delete test on tcp://10.192.122.236:2375...
2017-07-17T23:06:45-07:00 START: Running clone concurrent test...
2017-07-17T23:06:56-07:00 END: Running clone concurrent test...
--- PASS: TestConcurrency (26.94s)
	sanity_test.go:207: Successfully connected to tcp://10.192.122.236:2375
	sanity_test.go:207: Successfully connected to tcp://10.192.108.211:2375
PASS
coverage: 0.0% of statements
```

### CODE REVIEW

**Comment by venilnoronha** [comment]
The `fs.GetMountInfo(..)` implementation can easily be replaced with a one based on the `StackExchange/wmi` package. Here's a code snippet.

```go
import "github.com/StackExchange/wmi"

type Win32_Volume struct {
   DeviceID string
   Name     string
}

func ListVolumes() error {
    var volumes []Win32_Volume
    q := wmi.CreateQuery(&volumes, "")
    c := &wmi.Client{NonePtrZero: true, PtrNil: false, AllowMissingFields: true}
    if err := c.Query(q, &volumes); err != nil {
        return err
    }
    for _, v := range volumes {
       fmt.Println(v)
    }
    return nil
}
```

The output goes as follows.

```
{\\?\Volume{1cca2a47-0000-0000-0000-501f00000000}\ C:\}
{\\?\Volume{145c5662-0000-0000-0000-100000000000}\ C:\Users\Administrator\AppData\Local\docker-volume-vsphere\mounts\volName\}
```

I'm currently staying away from the `StackExchange/wmi` package as there is quite some bit of `PowerShell` code that we want migrated, and I'm not sure of the feasibility of achieving that using just this package. It would be good to first have a look at the options described in #1553 before we pick an approach.

**Comment by msterin** [comment]
>  I'm not sure of the feasibility of achieving that using just this package

WMI is WMI. If StackExchange provides access to WMI, they it's all doable from there. 
What are you suing in PS which is not WMI ?

**Comment by venilnoronha** [comment]
I'm talking about the code that watches device events in [`device_watcher_windows.go`](https://github.com/vmware/docker-volume-vsphere/blob/3a1ebbb302a5d1aae19c3dd91781176413816a83/client_plugin/utils/fs/device_watcher_windows.go#L35-L36). It's based on `WMI`; however, the `StackExchange/wmi` wrapper only provides an API for queries, and has no APIs that can do `Register-WmiEvent` or `Wait-Event`.

Also, I have a similar concern about the disk format code in [`fs_windows.go`](https://github.com/vmware/docker-volume-vsphere/blob/3a1ebbb302a5d1aae19c3dd91781176413816a83/client_plugin/utils/fs/fs_windows.go#L70-L74).

File: client_plugin/utils/fs/fs_linux.go [hunk 1]
```diff
@@ -354,3 +356,27 @@ func getDevicePath(volDev *VolumeDevSpec) (string, error) {
...
+
+// GetMountInfo returns a map of mounted volumes and devices.
+func GetMountInfo(mountRoot string) (map[string]string, error) {
+	volumeMountMap := make(map[string]string) // map [volume mount path] -> device
+	data, err := ioutil.ReadFile(linuxMountsFile)
+
+	if err != nil {
+		log.Errorf("Can't get info from %s (%v)", linuxMountsFile, err)
+		return volumeMountMap, err
+	}
```
Comment (shaominchen) [inline client_plugin/utils/fs/fs_linux.go]: Should we log the data for easier debugging, same as we have done in fs_windows.go?

File: client_plugin/utils/fs/fs_linux.go [hunk 2]
```diff
@@ -354,3 +356,27 @@ func getDevicePath(volDev *VolumeDevSpec) (string, error) {
...
+		if len(field) < 2 {
+			continue // skip empty line and lines too short to have our mount
+		}
+		// fields format: [/dev/sdb /mnt/vmdk/vol1 ext2 rw,relatime 0 0]
+		if filepath.Dir(field[1]) != mountRoot {
+			continue
+		}
+		volumeMountMap[filepath.Base(field[1])] = field[0]
+	}
+	return volumeMountMap, nil
```
Comment (shaominchen) [inline client_plugin/utils/fs/fs_linux.go]: Should we log the volumeMountMap for easier debugging, same as we have done in fs_windows.go?

Comment (venilnoronha) [inline client_plugin/utils/fs/fs_linux.go]: Addressed both comments in fb238fd.

**Comment by venilnoronha** [comment]
@msterin I'm merging this PR. I'll address additional comments separately, if any.

**Pull request merged**